ignite/metrics/ssim.py

```diff
@@ -116,7 +121,7 @@ def _gaussian_or_uniform_kernel(self, kernel_size, sigma):
 
     @reinit__is_reduced
     def update(self, output: Sequence[torch.Tensor]) -> None:
-        y_pred, y = output[0].detach(), output[1].detach()
+        y_pred, y = output[0].detach().to(self._device), output[1].detach().to(self._device)
```

Comment:
I think we should keep user's device until we update `self._sum_of_batchwise_ssim`.

Reply:
Sounds good, done!